tests/test-update-direct.c

```diff
@@ -246,6 +246,110 @@ test_cancel_update (EosUpdaterFixture *fixture,
   g_assert_cmpuint (helper.cancelled_error_count, ==, helper.cancel_calls_count);
 }
 
+static void
+update_version_state_changed_cb (EosUpdater *updater,
+                                 GParamSpec *pspec,
+                                 gpointer data)
+{
+  EosUpdaterState state = eos_updater_get_state (updater);
+  GMainLoop *loop = (GMainLoop *) data;
+
+  if (state == EOS_UPDATER_STATE_POLLING)
+    return;
+
+  g_main_loop_quit (loop);
+}
+
+static void
+test_update_version (EosUpdaterFixture *fixture,
+                     gconstpointer user_data)
+{
+  g_autoptr(GFile) server_root = NULL;
+  g_autoptr(EosTestServer) server = NULL;
+  g_autofree gchar *keyid = get_keyid (fixture->gpg_home);
+  g_autoptr(GError) error = NULL;
+  g_autoptr(EosTestSubserver) subserver = NULL;
+  g_autoptr(GFile) client_root = NULL;
+  g_autoptr(EosTestClient) client = NULL;
+  g_autoptr(EosUpdater) updater = NULL;
+  g_autoptr(GMainLoop) loop = g_main_loop_new (NULL, FALSE);
+  DownloadSource main_source = DOWNLOAD_MAIN;
+  const gchar *version = user_data != NULL ? (const gchar *) user_data : "";
```

Comment:
Nitpick: Brackets around the conditional makes the operator precedence clearer (just a code style issue to make it fit in with the rest of the code):
```
const gchar *version = (user_data != NULL) ? (const gchar *) user_data : "";
```